### ISSUE TITLE
Restrict line length to 80 characters (adjust eslint rule)

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-frontend#3045]
Code lines restricted to 80 characters lets you use many horizontal split screens in your editor which is nice.

```diff
diff --git a/.eslintrc b/.eslintrc
index c9d84498..6d8085f9 100644
--- a/.eslintrc
+++ b/.eslintrc
@@ -10,6 +10,13 @@
     "jest"
   ],
   "rules": {
+    "max-len": ["error", 80, 2, {
+      ignoreUrls: true,
+      ignoreComments: false,
+      ignoreRegExpLiterals: true,
+      ignoreStrings: false,
+      ignoreTemplateLiterals: false,
+    }],
     // FIXME: Deprecate use of findDOMNode https://github.com/mozilla/addons-frontend/issues/968
     "react/no-find-dom-node": "off",
   },
```

We discussed it here: https://github.com/mozilla/addons-frontend/pull/3036#discussion_r136206566

Unfortunately, `yarn eslint -- --fix` cannot help with this. 

**Please ask before fixing this issue as it will be hard to review and could introduce merge conflicts**.

**Comment by kumar303** [comment]
As a stepping stone we could maybe do `ignoreStrings: true` and `ignoreTemplateLiterals: true`.

**Comment by willdurand** [comment]
I think I mentioned it already, but maybe [Prettier](https://github.com/prettier/prettier) could help here. At least, it would avoid introducing bugs.

**Comment by muffinresearch** [comment]
@kumar303 this could be something worth considering prettier for as suggested by @willdurand.

**Comment by muffinresearch** [comment]
@willdurand snap! - I've played with it I think it's pretty good it really does format the code like we mostly have it. The only real nit is the requirement of a post-save / pre-commit hook.

**Comment by muffinresearch** [comment]
> The only real nit is the requirement of a post-save / pre-commit hook.

As long as we can either fail when files don't follow it / or have a bot add a commit having run it, it would probably not get in the way.

**Comment by muffinresearch** [comment]
FWIW: I started a patch to do 80chars and it was epic. On that basis alone, switching to use prettier would probably be quite a lot easier to implement.

**Comment by willdurand** [comment]
Well, we can maybe run prettier time to time? Given the intensity on code reviews, I dont think things will diverge too much from our standards.

**Comment by kumar303** [comment]
I'm kinda scared to see the initial patch that ports us to Prettier but I'm open to using it :) I've never really looked at it. I always assumed there would be some annoying little edge case where you want to format something a certain way and it won't let you.

**Comment by willdurand** [comment]
That's also a strength: no more personal opinion, the tool becomes the source of truth 😛

**Comment by muffinresearch** [comment]
Alright I'll put a patch together to see what it look like - that'll be a good way for everyone to see if they're on board with Prettier's output as a starting point.

**Comment by willdurand** [comment]
Fixed by #5313
